ITensorGPU/src/cuitensor.jl

```diff
@@ -2,8 +2,10 @@ import ITensors.NDTensors: NeverAlias, AliasStyle, AllowAlias
 import ITensors: ITensor
 import CUDA: CuArray
 
-function cuITensor(::Type{T}, inds::IndexSet) where {T<:Number}
-  return ITensor(Dense{float(T)}(CUDA.zeros(float(T), dim(inds))), inds)
+function cuITensor(eltype::Type{<:Number}, inds::IndexSet)
+  d = Dense(CUDA.zeros(eltype, dim(inds)))
```

Comment:
Can't we use `default_storagetype(CuVector{eltype})` here?

Reply:
Yes we can, and I have updated the code